src/compute/src/sink/persist_sink.rs

```diff
@@ -1035,9 +1083,19 @@ where
             for done_batch_metadata in done_batches.into_iter() {
                 in_flight_descriptions.remove(&done_batch_metadata);
 
-                let mut batches = in_flight_batches
+                let batch_set = in_flight_batches
                     .remove(&done_batch_metadata)
-                    .unwrap_or_else(Vec::new);
+                    .unwrap_or_default();
+
+                let mut batches = batch_set.finished;
+                if !batch_set.incomplete.is_empty() {
+                    let (lower, upper) = done_batch_metadata.clone();
```

Comment:
let's toss a consolidate in here, at least until we can revisit paul's change to consolidate in user batches